scripts/validate_links.py

```diff
@@ -0,0 +1,406 @@
+        pick = pick_next_url(domains, domain_to_urls, last_request_time)
+        if pick is None:
+            # All domains are on cooldown, sleep until the soonest one is ready
+            soonest = min(
+                (last_request_time[d] + REQUEST_DELAY for d in domains if domain_to_urls[d]),
+                default=time.time() + REQUEST_DELAY,
+            )
+            sleep_time = max(soonest - time.time(), 0.05)
+            time.sleep(sleep_time)
+            continue
+        domain, url = pick
+
+        # Classify and fetch
+        if url in url_cache:
+            print(f"[done {done_count + 1}/{total_urls}] {url} (cache hit)")
+            done_count += 1
+            continue
+        if not URL_VALIDATOR_REGEX.match(url):
+            url_cache[url] = "INVALID_URL"
+            print(f"[done {done_count + 1}/{total_urls}] {url} WARNING: Invalid URL format, not fetched.")
+            done_count += 1
+            continue
+
+        print(f"[done {done_count + 1}/{total_urls}] {url} ...", end=" ")
+        try:
+            resp = requests.get(
+                url, timeout=REQUEST_TIMEOUT, headers={"User-Agent": "django-components-link-checker/0.1"}
+            )
+            url_cache[url] = resp
+            print(f"{resp.status_code}")
+        except Exception as err:
+            url_cache[url] = err
+            print(f"ERROR: {err}")
+
+        last_request_time[domain] = time.time()
+        done_count += 1
+    return url_cache
+
+
+def check_fragment_in_html(html: str, fragment: str) -> bool:
+    """Return True if id=fragment exists in the HTML."""
+    print(f"Checking fragment {fragment} in HTML...")
+    soup = BeautifulSoup(html, "html.parser")
+    return bool(soup.find(id=fragment))
+
+
+def rewrite_url(url: str) -> Union[Tuple[None, None], Tuple[str, Union[str, re.Pattern]]]:
+    """Return (new_url, mapping_key) if a mapping applies, else (None, None)."""
+    for key, repl in URL_REWRITE_MAP.items():
+        if isinstance(key, str):
+            if url.startswith(key):
+                return url.replace(key, repl, 1), key
+        elif isinstance(key, re.Pattern):
+            if key.search(url):
+                return key.sub(repl, url), key
+        else:
+            raise ValueError(f"Invalid key type: {type(key)}")
+    return None, None
+
+
+def output_summary(errors: List[Tuple[str, int, str, str, str]], output: str):
+    # Format the errors into a table
+    headers = ["Type", "Details", "File", "URL"]
+    data = [
+        {"File": file + "#" + str(lineno), "Type": errtype, "URL": url, "Details": details}
+        for file, lineno, errtype, url, details in errors
+    ]
+    table = format_as_ascii_table(data, headers, include_headers=True)
+
+    # Output summary to file if specified
+    if output:
+        output_path = Path(output)
+        output_path.write_text(table + "\n", encoding="utf-8")
+    else:
+        print(table + "\n")
+
+
+# TODO: Run this as a test in CI?
```

Comment:
Altho it would be nice if this ran as part of CI, there's a lot of false positives. So this script makes more sense to ran once in couple of months to check that the URLs are still correct.